plugins/linear-genome-view/src/index.ts

```diff
@@ -142,6 +146,51 @@ export default class LinearGenomeViewPlugin extends Plugin {
         },
       })
     }
+
+    this.initFromLocString(pluginManager)
+  }
+
+  async initFromLocString(pluginManager: PluginManager) {
+    const parsed = queryString.parse(window.location.search)
+    if (pluginManager.rootModel && parsed.loc) {
+      if (!pluginManager.rootModel.session) {
+        // @ts-ignore types are not accurate here, they are stub abstracts
+        pluginManager.rootModel.setSession({ name: 'New session' })
+      }
+      const { session } = pluginManager.rootModel
+
+      // @ts-ignore types are not accurate here, they are stub abstracts
```

Comment:
there might be some type guards you can use here to help with these ts-ignores

Reply:
made some changes here so that it has less ts-ignores. it is difficult to get accurate inferences from all these Abstract types but made an attempt here